localstack/services/cloudwatch/cloudwatch_database_helper.py

```diff
@@ -0,0 +1,407 @@
+                "sum"	                NUMERIC,
+                "min"	                NUMERIC,
+                "max"	                NUMERIC,
+                PRIMARY KEY("id")
+            );
+            """
+            )
+            # create indexes
+            cur.executescript(
+                """
+            CREATE INDEX idx_single_metrics_comp ON SINGLE_METRICS (metric_name, namespace);
+            CREATE INDEX idx_aggregated_metrics_comp ON AGGREGATED_METRICS (metric_name, namespace);
+            """
+            )
+            conn.commit()
+
+    def add_metric_data(
+        self, account_id: str, region: str, namespace: str, metric_data: MetricData
+    ):
+        # TODO consider using thread-lock here instead of increasing busy-timeout
+        with sqlite3.connect(self.METRICS_DB, isolation_level="EXCLUSIVE") as conn:
+            conn.execute(
+                "PRAGMA busy_timeout = 20000"
+            )  # TODO check if we need to set timeout higher, testing with 20 seconds
+            cur = conn.cursor()
+
+            def _get_current_unix_timestamp_utc():
+                now = datetime.utcnow().replace(tzinfo=timezone.utc)
+                return int(now.timestamp())
+
+            for metric in metric_data:
+                unix_timestamp = (
+                    self._convert_timestamp_to_unix(metric.get("Timestamp"))
+                    if metric.get("Timestamp")
+                    else _get_current_unix_timestamp_utc()
+                )
+
+                inserts = []
+                if metric.get("Value") is not None:
+                    inserts.append({"Value": metric.get("Value"), "TimesToInsert": 1})
+                elif metric.get("Values"):
+                    counts = metric.get("Counts", [1] * len(metric.get("Values")))
+                    inserts = [
+                        {"Value": value, "TimesToInsert": int(counts[indexValue])}
+                        for indexValue, value in enumerate(metric.get("Values"))
+                    ]
+
+                for insert in inserts:
+                    for _ in range(insert.get("TimesToInsert")):
+                        cur.execute(
+                            self._get_insert_single_metric_query(),
+                            (
+                                account_id,
+                                region,
+                                metric.get("MetricName"),
+                                namespace,
+                                unix_timestamp,
+                                self._get_ordered_dimensions_with_separator(
+                                    metric.get("Dimensions")
+                                ),
+                                metric.get("Unit"),
+                                metric.get("StorageResolution"),
+                                insert.get("Value"),
+                            ),
+                        )
+
+                if statistic_values := metric.get("StatisticValues"):
```

Comment:
It would be great if we could also add a test where we try to insert single + static values at the same time (in one `put-metric-data` statement. I think it will throw an error.

Reply:
adding to follow-up task